### ISSUE TITLE
Parsing multi-line doctypes triggers a parser error

### ISSUE THREAD

**Issue by YorickPeterse** [YorickPeterse/oga#95]
HTML:

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"
    "http://www.w3.org/TR/html4/loose.dtd">
```

Script:

    ruby -r ./lib/oga -e 'p Oga.parse_xml(File.read("/tmp/test.html"))'

Produces:

```
/home/yorickpeterse/Private/Projects/ruby/oga/lib/oga/xml/parser.rb:271:in `parser_error': Unexpected end of input, expected doctype closing tag instead on line 1 (LL::ParserError)
        from /home/yorickpeterse/Private/Projects/ruby/oga/lib/oga/xml/parser.rb:247:in `each_token'
        from /home/yorickpeterse/Private/Projects/ruby/oga/lib/oga/xml/parser.rb:278:in `parse'
        from /home/yorickpeterse/Private/Projects/ruby/oga/lib/oga/xml/parser.rb:278:in `parse'
        from /home/yorickpeterse/Private/Projects/ruby/oga/lib/oga/oga.rb:12:in `parse_xml'
        from -e:1:in `<main>'
```

Lexer output:

```
[
    [0] [
        [0] :T_DOCTYPE_START,
        [1] nil,
        [2] 1
    ],
    [1] [
        [0] :T_DOCTYPE_NAME,
        [1] "HTML",
        [2] 1
    ],
    [2] [
        [0] :T_DOCTYPE_TYPE,
        [1] "PUBLIC",
        [2] 1
    ],
    [3] [
        [0] :T_STRING_DQUOTE,
        [1] nil,
        [2] 1
    ],
    [4] [
        [0] :T_STRING_BODY,
        [1] "-//W3C//DTD HTML 4.01 Transitional//EN",
        [2] 1
    ],
    [5] [
        [0] :T_STRING_DQUOTE,
        [1] nil,
        [2] 1
    ]
]
```

**Comment by binaryplease** [comment]
When will this fix be released?

**Comment by YorickPeterse** [comment]
I'll prepare a release for this later today.

**Comment by YorickPeterse** [comment]
@binaryplease This has been released as version 0.3.3 which is being pushed to RubyGems as we speak.
